target/linux/ath79/dts/qca9563_tplink_archer-x6-v2.dtsi

```diff
@@ -0,0 +1,170 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "qca956x.dtsi"
+
+/ {
+	chosen {
+		bootargs = "console=ttyS0,115200n8";
+	};
+
+	aliases {
+		led-boot = &power;
+		led-failsafe = &power;
+		led-running = &power;
+		led-upgrade = &power;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		power: power {
+			label = "tp-link:green:power";
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+			default-state = "on";
+		};
+
+		wlan2g {
+			label = "tp-link:green:wlan2g";
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy1tpt";
+		};
+
+		wlan5g {
+			label = "tp-link:green:wlan5g";
+			gpios = <&gpio 19 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0tpt";
+		};
+
+		lan {
+			label = "tp-link:green:lan";
+			gpios = <&gpio 20 GPIO_ACTIVE_LOW>;
+		};
+
+		wan {
+			label = "tp-link:green:wan";
+			gpios = <&gpio 7 GPIO_ACTIVE_LOW>;
+		};
+
+		wan_fail {
+			label = "tp-link:amber:wan";
+			gpios = <&gpio 8 GPIO_ACTIVE_LOW>;
+		};
+
+		wps {
+			label = "tp-link:green:wps";
+			gpios = <&gpio 9 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "Reset button";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 5 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+
+		wps {
+			label = "WPS button";
+			linux,code = <KEY_WPS_BUTTON>;
+			gpios = <&gpio 6 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+	};
+};
+
+&pcie {
+	status = "okay";
+};
+
+&uart {
+	status = "okay";
+};
+
+&gpio {
+	status = "okay";
+};
+
+&spi {
+	status = "okay";
+	num-cs = <1>;
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <25000000>;
+
+		partitions {
```

Comment:
+1 For creating a shared DTSI for the C6 and A6!

I suppose the partition layout of the Archer A6 also differs like it is with the Archer C7 v5 and Archer A7 v5, can you check that? The partition layout should be readable from the TP-Link firmware upgrade files for both devices.

In case they differ, the layout should be defined in the respective board dts.

Reply:
I can't find any A6(US) vendor firmware binaries. Looks like they haven't released one yet.
I can move the C6 specific leds/keys out of the dtsi along with the mtd partitions
into the C6 dts. When the A6 vendor firmware is out, we can add a A6 dts with its
own leds/key/mtdparts. If that makes sense I'll update the pull request.

Reply:
Yup they seem to not have been released (yet?). LEDs and Keys are named generic "tp-link" so they can be kept in the shared dtsi. Just mtdparts are highly likely to differ.

Reply:
I just submitted pull request for C6 v2 (US), it may be compatible for A6 (US) and A6 (TW), https://github.com/openwrt/openwrt/pull/2470

Reply:
> 
> I just submitted pull request for C6 v2 (US), it may be compatible for A6 (US) and A6 (TW), #2470

I'd like to port it to C6 v2 (JP).
Could you explain quickly what do I need to check in the binary? Small screenshots so that I can port it and test.